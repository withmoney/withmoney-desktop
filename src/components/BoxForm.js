import React from 'react';
import PropTypes from 'prop-types';

const BoxForm = ({
  title,
  subtitle,
  fields,
  footer,
  onSubmit,
}) => (
  <div className="box-form">
    <h1 className="box-form__title">{title}</h1>
    <form className="box-form__form" onSubmit={onSubmit}>
      <h3 className="box-form__subtitle">{subtitle}</h3>
      {fields}
      <div className="box-form__footer">
        {footer}
      </div>
    </form>
  </div>
);

BoxForm.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  fields: PropTypes.any.isRequired,
  footer: PropTypes.any.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default BoxForm;
