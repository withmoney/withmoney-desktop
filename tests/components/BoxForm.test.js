import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import BoxForm from '../../src/components/BoxForm';

describe('BoxForm', () => {
  let Component;
  const defaultProps = {
    title: 'Title',
    subtitle: 'Subtitle',
    fields: <input name="input" />,
    footer: <button type="submit">Action</button>,
    onSubmit: jest.fn(),
  };

  beforeAll(() => {
    Component = global.setup(BoxForm, defaultProps);
  });

  it('should render a component correctly', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should call action when form is submitted', () => {
    const wrapper = mount(Component());

    wrapper.find('form').simulate('submit');

    expect(defaultProps.onSubmit).toBeCalled();
  });
});
