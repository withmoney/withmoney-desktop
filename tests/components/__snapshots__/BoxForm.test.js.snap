// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxForm should render a component correctly 1`] = `
<div
  className="box-form"
>
  <h1
    className="box-form__title"
  >
    Title
  </h1>
  <form
    className="box-form__form"
    onSubmit={[MockFunction]}
  >
    <h3
      className="box-form__subtitle"
    >
      Subtitle
    </h3>
    <input
      name="input"
    />
    <div
      className="box-form__footer"
    >
      <button
        type="submit"
      >
        Action
      </button>
    </div>
  </form>
</div>
`;
